src/pk/ecc/ecc_get_curve_by_name.c

```diff
@@ -11,6 +11,185 @@
 
 #ifdef LTC_MECC
 
+const struct {
+   const char *OID;
+   const char *names[6];
+} _curve_names[] = {
+#if defined(LTC_ECC_SECP112R1) || defined(LTC_ECC112)
+   {
+      "1.3.132.0.6", { "SECP112R1", "ECC-112", NULL }
```

Comment:
Can you please use the official spelling? (as proposed in https://github.com/libtom/libtomcrypt/pull/236#discussion_r176721275)

Reply:
Do you mean:
```
secp256r1
NIST P-256
prime256v1
ECC-256
P-256
```
?

I also want to point out that apart from official names there is something like "industry standard" how these curves are named in other libraries - e.g. openssl:
```
$ openssl ecparam -list_curves | grep 'prime field'
  secp112r1 : SECG/WTLS curve over a 112 bit prime field
  secp112r2 : SECG curve over a 112 bit prime field
  secp128r1 : SECG curve over a 128 bit prime field
  secp128r2 : SECG curve over a 128 bit prime field
  secp160k1 : SECG curve over a 160 bit prime field
  secp160r1 : SECG curve over a 160 bit prime field
  secp160r2 : SECG/WTLS curve over a 160 bit prime field
  secp192k1 : SECG curve over a 192 bit prime field
  secp224k1 : SECG curve over a 224 bit prime field
  secp224r1 : NIST/SECG curve over a 224 bit prime field
  secp256k1 : SECG curve over a 256 bit prime field
  secp384r1 : NIST/SECG curve over a 384 bit prime field
  secp521r1 : NIST/SECG curve over a 521 bit prime field
  prime192v1: NIST/X9.62/SECG curve over a 192 bit prime field
  prime192v2: X9.62 curve over a 192 bit prime field
  prime192v3: X9.62 curve over a 192 bit prime field
  prime239v1: X9.62 curve over a 239 bit prime field
  prime239v2: X9.62 curve over a 239 bit prime field
  prime239v3: X9.62 curve over a 239 bit prime field
  prime256v1: X9.62/SECG curve over a 256 bit prime field
  wap-wsg-idm-ecid-wtls6: SECG/WTLS curve over a 112 bit prime field
  wap-wsg-idm-ecid-wtls7: SECG/WTLS curve over a 160 bit prime field
  wap-wsg-idm-ecid-wtls8: WTLS curve over a 112 bit prime field
  wap-wsg-idm-ecid-wtls9: WTLS curve over a 160 bit prime field
  wap-wsg-idm-ecid-wtls12: WTLS curvs over a 224 bit prime field
  brainpoolP160r1: RFC 5639 curve over a 160 bit prime field
  brainpoolP160t1: RFC 5639 curve over a 160 bit prime field
  brainpoolP192r1: RFC 5639 curve over a 192 bit prime field
  brainpoolP192t1: RFC 5639 curve over a 192 bit prime field
  brainpoolP224r1: RFC 5639 curve over a 224 bit prime field
  brainpoolP224t1: RFC 5639 curve over a 224 bit prime field
  brainpoolP256r1: RFC 5639 curve over a 256 bit prime field
  brainpoolP256t1: RFC 5639 curve over a 256 bit prime field
  brainpoolP320r1: RFC 5639 curve over a 320 bit prime field
  brainpoolP320t1: RFC 5639 curve over a 320 bit prime field
  brainpoolP384r1: RFC 5639 curve over a 384 bit prime field
  brainpoolP384t1: RFC 5639 curve over a 384 bit prime field
  brainpoolP512r1: RFC 5639 curve over a 512 bit prime field
  brainpoolP512t1: RFC 5639 curve over a 512 bit prime field
```
We should be IMO able to understand these "common" names.

Reply:
sounds good

Reply:
btw regarding the names

```
secp256r1 (secg name)
NIST P-256 (safecurves uses this name, but fips-186-4 calls it "Curve P-256" so should we add that as well?)
prime256v1 (is that the X9.62 name?)
ECC-256 (the old libtom name)
P-256 (where does this come from?)
```

this comment is only relevant if we add `ecc_curve_get_name()` ... I don't like the inconsistency in the openssl naming...

```
...
  secp521r1 : NIST/SECG curve over a 521 bit prime field
  prime192v1: NIST/X9.62/SECG curve over a 192 bit prime fie
...
  wap-wsg-idm-ecid-wtls6: SECG/WTLS curve over a 112 bit prime field
...
```

I'd propose to use the names if available in this order of the standards bodies: NIST-SECG-ANSI-BRAINPOOL-OTHERS

Reply:
Ad P-256 & co. - these names are also used in JSON Web Signature (JWS) related RFCs 7515, 7518. It is perhaps based on curve names from https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.186-4.pdf

Reply:
Do we really need something like `ecc_curve_get_name()`? What would be the use case?

Reply:
> What would be the use case?

A CLI tool similar to `openssl ecparam -list_curves | grep 'prime field'`?
We can also leave it as it is for now and add the API as soon as that tool is written ;)